Monika After Story/game/zz_weather.rpy

```diff
@@ -780,14 +779,42 @@ init 800 python:
         IN:
             new_weather - weather to change to
             by_user - flag for if user changes weather or not
+            set_persistent - whether or not we want to make this weather persistent
         """
 
         if by_user is not None:
             mas_weather.force_weather = bool(by_user)
 
+        if set_persistent:
+            persistent._mas_current_weather = new_weather.weather_id
+
         mas_current_weather.exit(new_weather)
         mas_setWeather(new_weather)
 
+    def mas_startupWeather():
+        """
+        Runs a weather startup alg, checking whether or not persistent weather should be used
+        Sets weather accordingly
+        """
+        #If the current bg doesn't support weather, we'll just return
+        if store.mas_current_background.disable_progressive:
+            return
+
+        elif not store.mas_weather.force_weather and not store.skip_setting_weather:
```

Comment:
the if above returns so dont need to elif this.

in fact since I don't see any other code in the same line as this elif, the inverse of this condition can just be added to the starting if statement.